src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -964,4 +895,101 @@ public void setNumOfRubricSubQuestions(int numOfRubricSubQuestions) {
         return rubricSubQuestions;
     }
 
+    /**
+     * Calculates the response frequency, percentage frequency and average value
+     * for each choice and each sub-question given the responses and question details
+     *
+     * For responseFrequency, last element in each row stores the total number of responses for the sub-question.
+     * e.g.
+     * responseFrequency[subQuestionIndex][choiceIndex]
+     * -> is the number of times choiceIndex is chosen for subQuestionIndex
+     * responseFrequency[subQuestionIndex][numOfRubricChoices]
+     * -> is the total number of the responses for the given sub-question
+     *
+     * For percentageFrequencyAndAverage, values are set to 0 if there are no responses to that sub-question.
+     * Average value is set to 0 if there are no assigned weights.
+     * e.g.
+     * percentageFrequencyAndAverageValue[subQuestionIndex][choiceIndex]
+     * -> is the percentage choiceIndex is chosen for subQuestionIndex
+     * percentageFrequencyAndAverageValue[subQuestionIndex][numOfRubricChoices]
+     * -> is the average weight of the responses for the given sub-question
+     */
```

Comment:
This is too detailed for a header comment. To explain the class, something like "Statistics for a rubric question, consisting of ...".
The explanations are more apt for the variables:
```java
/**
 * Last element in each row ...
 * ... the given sub-question.
 */
int[][] responseFrequency;

/**
 * Values are set to 0 ...
 * ... the given sub-question.
 */
float[][] percentageFrequencyAndAverage;
```